### PR TITLE
Added RailFence cipher

### PR DESCRIPTION
**Pull request by yogesh1997**
Updated and Added New RailFence cipher

* crypto
   * Added railfence cipher

### CODE REVIEW

**Comment by jksuom** [comment]
There is trailing whitespace on line 2259 in crypto.py. That should be removed.

Please do not open another PR to update this. Just add new commits by pushing to this same branch without closing.

**Comment by yogesh1997** [comment]
Removed Trailing whitespaces

**Comment by yogesh1997** [comment]
> There is trailing whitespace on line 2259 in crypto.py. That should be removed.
> 
> Please do not open another PR to update this. Just add new commits by pushing to this same branch without closing.
Removed the whitespaces

**Comment by yogesh1997** [comment]
> There is trailing whitespace on line 2259 in crypto.py. That should be removed.
> 
> Please do not open another PR to update this. Just add new commits by pushing to this same branch without closing.

>Removed the whitespaces

**Comment by yogesh1997** [comment]
Removed the whitespaces

File: sympy/crypto/tests/test_crypto.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
       encipher_elgamal, decipher_elgamal, dh_private_key, dh_public_key,
       dh_shared_key, decipher_shift, decipher_affine, encipher_bifid,
       decipher_bifid, bifid_square, padded_key, uniq, decipher_gm,
-      encipher_gm, gm_public_key, gm_private_key)
+      encipher_gm, gm_public_key, gm_private_key,encipher_railfence,decipher_railfence)
```
Comment (oscargus) [inline sympy/crypto/tests/test_crypto.py]: space after ,

**Comment by czgdp1807** [comment]
LGTM. @yogesh1997 Are you interested in taking this forward and completing it.

**Comment by czgdp1807** [comment]
@yogesh1997 Please add `Closes https://github.com/sympy/sympy/pull/15789` in [this comment](https://github.com/sympy/sympy/pull/15534#issue-233110216).

**Pull request closed without merging**